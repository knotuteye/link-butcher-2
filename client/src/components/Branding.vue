<template>
  <div class="brand slide-in-top">
    <h1 class="masthead">
      {{ name }}
    </h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class Branding extends Vue {
  @Prop() private name!: string
}
</script>

<style scoped>
h1.masthead {
  color: #ffffff;
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
  font-size: 4em;
  text-shadow: 1px 5px 10px #200053a2;
  letter-spacing: 4px;
  text-transform: lowercase;
  text-align: center;
}

/* Animation */

.slide-in-top {
  animation: slide-in-top 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@keyframes slide-in-top {
  0% {
    transform: translateY(-1000px);
    opacity: 0;
  }
  100% {
    transform: translateY(0);
    opacity: 1;
  }
}
</style>
