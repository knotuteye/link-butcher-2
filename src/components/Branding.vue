<template>
  <div class="brand">
    <h1 class="masthead">
      {{ name }}
    </h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class Branding extends Vue {
  @Prop() private name!: string
}
</script>

<style scoped>
h1.masthead {
  color: #ffffff;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 500;
  font-size: 2em;
  text-transform: lowercase;
}

div.brand {
  border: 2px solid #ffffff69;
  width: max-content;
  padding: 1rem 2rem;
}
</style>
