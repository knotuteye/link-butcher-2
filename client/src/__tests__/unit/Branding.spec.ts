import { shallowMount } from '@vue/test-utils'
import Branding from '../../components/Branding.vue'

describe('Branding.vue', () => {
  test('renders props.name when passed', () => {
    const name = 'pbid.io'
    const wrapper = shallowMount(Branding, {
      propsData: { name }
    })
    expect(wrapper.text()).toMatch(name)
  })

  test('renders as designed', () => {
    const name = 'pbid.io'
    const wrapper = shallowMount(Branding, {
      propsData: { name }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
