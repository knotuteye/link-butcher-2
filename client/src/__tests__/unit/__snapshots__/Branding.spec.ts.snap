// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branding.vue renders as designed 1`] = `
<div
  class="brand slide-in-top"
>
  <h1
    class="masthead"
  >
    
    pbid.io
  
  </h1>
</div>
`;
